src/utils/hashPassword.ts

```diff
@@ -0,0 +1,29 @@
+export async function hashPasswordPBKDF2(password: string, salt: Uint8Array): Promise<string> {
+	const encoder = new TextEncoder();
+	const passwordBuffer = encoder.encode(password);
+
+	// Import password as key material
+	const key = await crypto.subtle.importKey(
+		'raw',
+		passwordBuffer,
+		{ name: 'PBKDF2' },
+		false,
+		['deriveBits']
+	);
+
+	// Derive key using PBKDF2-HMAC-SHA256
+	const derivedKey = await crypto.subtle.deriveBits(
+		{
+			name: 'PBKDF2',
+			salt,
+			iterations: 600000,
+			hash: 'SHA-256'
+		},
+		key,
+		256
+	);
+
+	// Convert derived key to Base64
+	const derivedKeyBase64 = btoa(String.fromCharCode(...new Uint8Array(derivedKey)));
```

Comment:
If we are using the node compatibility layer, I would try doing this:
```suggestion
	const derivedKeyBase64 = new Uint8Array(derivedKey).toString('base64');
```

Check if this works

Reply:
Yeah, I found we already added `compatibility_flags = ["nodejs_compat"]` in `wrangler.toml`, but wondering how to use `Buffer.from().toString('base64')` in refactored version..

Reply:
Just try it in a test endpoint and see if it breaks, can be a random test buffer 🤷‍♂️ from the string "test" or something